app/questionnaire/navigator.py

```diff
@@ -199,8 +191,12 @@ def get_blocks(self):
         blocks = []
 
         for group_index, group in enumerate(SchemaHelper.get_groups(self.survey_json)):
-            no_of_repeats = 1
+            if 'skip_condition' in group:
```

Comment:
if we are putting this on groups maybe we want to support blocks too? would have made some of the census routing rule a bit simpler?